libdnf/comps/environment/query.cpp

```diff
@@ -63,25 +65,29 @@ EnvironmentQuery::EnvironmentQuery(const BaseWeakPtr & base) : base(base) {
             continue;
         }
 
+        repoid = solvable->repo->name;
+
         // Add installed environments directly, because there is only one solvable for each
-        if (strcmp(solvable->repo->name, "@System") == 0) {
+        if (strcmp(repoid, "@System") == 0) {
```

Comment:
If `repoid` was `std::string_view`, you could do `repoid == "@System"` here, which is a bit nicer in C++.